hkmc2/shared/src/test/mlscript/codegen/SelfReferences.mls

```diff
@@ -28,9 +28,8 @@ Foo.self
 :e
 module Foo with
   val self = this
-//│ ╔══[ERROR] Functions returning module values must have explicit return types.
-//│ ║  l.30: 	  val self = this
-//│ ╙──      	      ^^^^
```

Comment:
How come we lost this location?

Reply:
It used to check the term definitions, so the location is from the method. Now it checks the terms themselves. so the location is from the keyword `this`, and `this` doesn't come with a location 🤕